src/index.js

```diff
@@ -26,57 +28,41 @@ ffmpeg.setFfmpegPath(require('ffmpeg-static').path);
  * @returns A fluent ffmpeg process - has pipe() method.
  */
 module.exports = function (filename, opts) {
-  ['height', 'width', 'fps'].forEach(key => {
-    assert(!isNaN(opts[key]), `${key} should be number - got ${opts[key]}`);
-  });
-  const { start = 0 } = opts;
-  const filters = [
+  ow(opts, ow.object.partialShape({
+    fps: ow.number.integer,
+    height: ow.number.integer,
+    width: ow.number.integer
+  }));
+  const { fps, height, loop = true, start = 0, width } = opts;
+  const filters = loop ? [
     { filter: 'concat', options: { n: 2, v: 1, a: 0 }, outputs: 'concat' },
     {
       filter: 'setpts',
       inputs: 'concat',
       options: 'N/(FRAME_RATE*TB)'
     }
-  ];
+  ] : [];
   const command = ffmpeg()
     // Using -ss and -stream_loop together does not work well, so we have a
     // single non-looped version first to seek on.
     .input(filename)
-    .inputOption('-ss', start)
-    .input(filename)
-    .inputOption('-stream_loop', -1)
+    .inputOption('-ss', start);
+  if (loop) command.input(filename).inputOption('-stream_loop', -1);
+  command
     .noAudio()
     .outputFormat('rawvideo')
     .outputOption('-vcodec', 'rawvideo')
     .outputOption('-pix_fmt', 'rgba')
-    .outputOption('-s', `${opts.width}x${opts.height}`)
-    .outputOption('-r', opts.fps);
-  applyCrop(filters, opts);
-  command.complexFilter(filters);
+    .outputOption('-s', `${width}x${height}`)
+    .outputOption('-r', fps);
+  const crop = cropFilter(opts);
+  if (crop) {
```

Comment:
I know we aren't a big fan of mutating things but I'm not sure this is any better than the old `applyCrop` version... It really isn't any easier to understand. If we do want to do things this way then maybe create another `appendFilter` function to do this bit instead? We could even use that for some of the other above filters if you felt so inclined as well.

```js
const filters = [];
if (loop) appendFilter(filters, loopFilter);
if (crop) appendFilter(filters, crop);
```